scripts/extensions/ai-widget/src/ai-assistant.tsx

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+import {IArticleSideWidgetComponentType} from 'superdesk-api';
+import {Spacer} from 'superdesk-ui-framework/react';
+import {superdesk} from './superdesk';
+import {configuration} from './configuration';
+import getHeadlinesWidget from './headlines/headlines-widget';
+import getSummaryWidget from './summary/summary-widget';
+import DefaultAiAssistantPanel from './main-panel';
+
+export type IAiAssistantSection = 'headlines' | 'summary' | null;
+
+interface IState {
+    activeSection: IAiAssistantSection;
+
+    /**
+     * Handle loading of each request separately,
+     */
+    loadingHeadlines: boolean;
+    loadingSummary: boolean;
+
+    headlines: Array<string>;
+    error: boolean;
+    summary: string;
+}
+
+export class AiAssistantWidget extends React.PureComponent<IArticleSideWidgetComponentType, IState> {
+    constructor(props: IArticleSideWidgetComponentType) {
+        super(props);
+
+        this.state = {
+            activeSection: null,
+            headlines: [],
+            error: false,
+            loadingSummary: true,
+            loadingHeadlines: true,
+            summary: '',
+        };
+
+        this.setError = this.setError.bind(this);
+        this.generateHeadlines = this.generateHeadlines.bind(this);
+        this.generateSummary = this.generateSummary.bind(this);
+    }
+
+    setError() {
+        this.setState({
+            error: true,
+        });
+    }
+
+    generateHeadlines() {
+        if (configuration.generateHeadlines == null) {
+            this.setError();
```

Comment:
Headlines are only generated from headlines widget. And we don't show headlines widget if we don't have the configuration. Then it means this line would never execute - so let's drop it. Same with `generateSummary`